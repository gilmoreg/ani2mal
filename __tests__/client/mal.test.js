const fetchMock = require('fetch-mock');
const Mal = require('../../assets/mal');
const fakes = require('../../helpers/fakes');

describe('Client side MAL', () => {
  beforeEach(() => {
    // Throw error on any fetch calls that aren't mocked
    fetchMock.catch(500);
  });

  afterEach(() => {
    fetchMock.restore();
  });

  it('check with valid credentials', (done) => {
    fetchMock.mock('*', fakes.malValidCheckResponse);
    Mal.check('test', 'test')
    .then((res) => {
      expect(res).toEqual(true);
      done();
    });
  });

  it('check with invalid credentials', (done) => {
    fetchMock.mock('*', fakes.malInvalidCheckResponse);
    Mal.check('test', 'test')
    .then((res) => {
      expect(res).toEqual(false);
      done();
    });
  });

  it('add should add an anime', (done) => {
    fetchMock.mock('*', fakes.malAddSuccess);
    Mal.add('auth', 'test')
    .then((res) => {
      expect(res).toMatchSnapshot();
      done();
    });
  });

  it('should handle an error from Google Cloud proxy', (done) => {
    fetchMock.mock('*', { throws: 'Error' });
    Mal.check('test', 'test')
    .then((res) => {
      expect(res).toMatchSnapshot();
      done();
    });
  });

  it('should handle an error from MAL', (done) => {
    fetchMock.mock('*', { throws: 'Error' });
    Mal.add('auth', 'test')
    .then((res) => {
      expect(res).toMatchSnapshot();
      done();
    });
  });
});
