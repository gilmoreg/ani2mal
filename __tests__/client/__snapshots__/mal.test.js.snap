// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client side MAL add should add an anime 1`] = `"Created"`;

exports[`Client side MAL should handle an error from Google Cloud proxy 1`] = `[Error: Error]`;

exports[`Client side MAL should handle an error from MAL 1`] = `[Error: Error]`;
